map/bookmark_manager.cpp

```diff
@@ -83,7 +83,7 @@ std::string GetFileNameForExport(BookmarkManager::KMLDataCollectionPtr::element_
   return fileName;
 }
 
-BookmarkManager::SharingResult ExportSingleFile(BookmarkManager::KMLDataCollectionPtr::element_type::value_type const & kmlToShare)
+BookmarkManager::SharingResult ExportSingleFileKml(BookmarkManager::KMLDataCollectionPtr::element_type::value_type const & kmlToShare)
```

Comment:
1. What is the value of copy-pasting the code into a new function? Let's extend this one instead and add switch inside.
2. Let's support exporting KML, KMZ and GPX if it's easy.

Reply:
1. Export to GPX doesn't have export zip and cleanup of temporary file, so about half of the code is different. What benefit we have if we put the same code into switch branches - now each function covers specific scenario and if we merge them we need to use `if` several time to adjust logic?
2. I had an idea to add KMZ enum value to distinguish between export of single track (Text) and export all (Kmz). But because we split export/export all by number of collections I removed it for now (until there is no 'Export all to Gpz')

Reply:
What happens with that temporary file? Is it cleaned later by the system somehow?

Reply:
> What happens with that temporary file? Is it cleaned later by the system somehow?

It works this way:
1. It creates file like "/data/user/0/app.organicmaps.debug/cache/Route.kml.tmp505640107344"
2. File is removed by OM code and kmz file is created
3. It shares file like "/data/user/0/app.organicmaps.debug/cache/Route.kmz"
4. In Android studio file browser I see folder /data/data/app.organicmaps.debug/cache on my phone after sharing is done with several tracks including the latest one "Route.kmz"

In this folder I see some tracks for previous days also. I was not able to find cleanup code for this folder in OM code, based on this answer https://stackoverflow.com/questions/9942560/when-to-clear-the-cache-dir-in-android application should do some housekeeping for it. But better to check it with our Android experts, I could be wrong. I tried to check this folder for non-debug build of OM that I have on my phone, but unfortunately there is an error 

![image](https://github.com/organicmaps/organicmaps/assets/128428520/21c94238-be00-4bbc-9f40-205effffa26c)

Reply:
Interesting, so we have some temporary files stuck there indefinitely... @rtsisyk @Jean-BaptisteC is there a way to know when a shared file was accessed/read by some other app and can be safely deleted? What Android guidelines say about it?

P.S. It's better to test beta or web flavor for non-debug builds.

Reply:
If the cache can be cleaned by the system or by a user, then let's leave it as is.